src/storage/storage.ts

```diff
@@ -78,6 +84,27 @@ export class Storage implements IStorage {
     this.version = version;
   }
 
+  /**
+   * Set the tx signing function
+   * @param {EcdsaTxSign} txSign The signing function
+   */
+  setTxSign(txSign: EcdsaTxSign): void {
+    this.txSignFunc = txSign;
```

Comment:
question: Would it be too much of an effort to create a `txSignFunc` and test it? The best case would be on an integration test, but I believe just a unit test would be enough for now.

That way we would ensure the override process is actually working.

Reply:
`getSignatureForTx` is the perfect example, it fetches/derives the mainKey and uses the storage to iterate on the inputs that are owned by the wallet, signing only those that belong to the wallet.

Any other test method would essentially need to have to do the same.
During tests what I would do is create a function that prints something to stdout and call `getSignatureForTx`, so I would know it called the correct method and that it works as expected.

Reply:
I'm not sure if I understand: `getSignatureForTx` is an example of signing function - actually the default one used by the lib, right?

CoveCov is informing that we have not tested this line `92`. As I understand, to increase this coverage in unit tests we would just need to call `setTxSign` with a mocked function and ensure it's actually invoked whenever a signature is required within the scope of this storage/wallet.

Integration tests for an actual valid external signature could be left to another moment.

Reply:
Created an integration test showcasing this.
It will also increase the coverage since it tests the custom signature method case.